src/components/forms/sendEmailForm/index.tsx

```diff
@@ -1,53 +1,112 @@
-import React from 'react';
-import { Form, Input, message } from 'antd';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Form, Input, Switch, message } from 'antd';
 import { SubmitButton } from '../../../components/themedComponents';
 import ProtectedApiClient from '../../../api/protectedApiClient';
 import {
   SendEmailFormValues,
   SendEmailRequest,
 } from '../../../components/forms/ducks/types';
 import { requiredRule } from '../../../utils/formRules';
+import { site } from '../../../constants';
+import { useTranslation } from 'react-i18next';
+import { n } from '../../../utils/stringFormat';
+import DOMPurify from 'isomorphic-dompurify';
+
+const PreviewSwitch = styled(Switch)`
+  display: flex;
+  align-items: center;
+  margin-bottom: 10px;
+  padding: 10px 5px;
+`;
+
+const EmailPreview = styled.div`
+  min-height: 250px;
+  margin-bottom: 24px;
+  max-height: 800px;
+  border: 1px solid #d9d9d9;
+  border-radius: 4px;
+  padding: 3px 11px;
+  resize: vertical;
+  overflow-y: scroll; // required for resizing
+`;
 
 interface SendEmailFormProps {
   readonly emails: string[];
 }
 
 const SendEmailForm: React.FC<SendEmailFormProps> = ({ emails }) => {
+  const { t } = useTranslation(n(site, ['forms']), {
+    keyPrefix: 'volunteer_emailer',
+    nsMode: 'fallback',
+  });
+
   const [sendEmailForm] = Form.useForm();
 
+  const [showPreview, setShowPreview] = useState<boolean>(false);
+  const [bodyContent, setBodyContent] = useState<string>('');
+  const [sanitizedBodyContent, setSanitizedBodyContent] = useState<string>('');
+
+  const togglePreview = (isShowPreview: boolean) => {
+    setShowPreview(isShowPreview);
+    if (isShowPreview) setSanitizedBodyContent(DOMPurify.sanitize(bodyContent));
+  };
+
   const onFinishSendEmail = (values: SendEmailFormValues) => {
+    if (emails.length === 0) {
+      message.error(t('users_required'));
+      return;
+    }
+
     const sendEmailRequest: SendEmailRequest = {
       ...values,
       emails,
     };
 
     ProtectedApiClient.sendEmail(sendEmailRequest)
       .then(() => {
-        message.success('Email sent!');
+        message.success(t('success'));
       })
       .catch((err) =>
-        message.error(`Email could not be sent: ${err.response.data}`),
+        message.error(t('response_error', { error: err.response.data })),
       );
   };
 
   return (
-    <Form name="sendEmail" form={sendEmailForm} onFinish={onFinishSendEmail}>
+    <Form
+      name="sendEmail"
+      form={sendEmailForm}
+      onFinish={onFinishSendEmail}
+      onValuesChange={(changedValues, _) => {
+        if (changedValues.emailBody !== undefined)
```

Comment:
if the emailBody is undefined, would we want to prompt the user to do anything? or just ignore until they input something that's valid.

Reply:
Good question! In this case, `emailBody` would be undefined only when the user edits a field that's not the body, since `changedValues` only contains fields that were modified when the user inputs something. This check is just to make sure we don't accidentally clear the body when the user modifies the subject.